### PR TITLE
Added Books: Competitive Programming and Introduction to Algorithms

### PR DESCRIPTION
**Pull request by PedrV**
## Hacktoberfest notes:

- due to volume of submissions, we may not be able to review PRs that do not pass tests and do not have informative titles.
- please read our [contributing guidelines](/CONTRIBUTING.md)
- be sure to check the output of Travis-CI for linter errors
- if this is your first open source contribution, make sure it's not your last!

## What does this PR do?
Add Resource(s)

### Description 
Added 2 books Competitive Programming by S Halim, F Halim, and Introduction to Algorithms, TH Cormen, CE Leiserson, RL Rivest and C Stein.

### Why is this valuable (or not)?
Introduction to Algorithms is an absolute fundamental book.
Competitive Programming  is a great book to begin competitive programming.

### How do we know it's really free?
Books made available by the author.

### For book lists, is it a book? For course lists, is it a course? etc.
Book.

### Checklist:
- [x] Not a duplicate
- [x] Included author(s) if appropriate
- [ ] Lists are in alphabetical order
- [ ] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -346,6 +346,7 @@
 * [The Great Tree List Recursion Problem](http://cslibrary.stanford.edu/109/TreeListRecursion.pdf) (PDF)
 * [The Kademlia Protocol Succinctly](https://www.syncfusion.com/ebooks/kademlia_protocol_succinctly) - Marc Clifton
 * [Think Complexity](http://greenteapress.com/complexity/) (PDF)
+* [Introduction to Algorithms, Second Edition](https://web.ist.utl.pt/~fabio.ferreira/material/asa/clrs.pdf) - Thomas H. Cormen et al. (PDF)
```
Comment (eshellman) [inline free-programming-books.md:349]: how do we know this is authorized distribution?

File: free-programming-books.md [hunk 2]
```diff
@@ -379,6 +380,7 @@
 
 * [Competitive Programmer's Handbook](https://cses.fi/book.html) - Antti Laaksonen (PDF)
 * [Competitive Programming, 1st Edition](https://cpbook.net/#CP1details) (PDF)
+* [Competitive Programming, 2nd Edition](https://www.comp.nus.edu.sg/~stevenha/myteaching/competitive_programming/cp2.pdf) - S Halim, F Halim (PDF)
```
Comment (eshellman) [inline free-programming-books.md:383]: also does not appear to be an authorized distribution

Comment (PedrV) [reply free-programming-books.md:383]: This is an authorized distribution, the second version of the book is now retired and the author made it available in the website cpbook.net, it's easy to check... I just used a different link ...

File: free-programming-books.md (hunk 1 above)
Comment (PedrV) [reply free-programming-books.md:349]: Even though I thought that was publicly available, after some search, I cannot confirm this statement.

File: free-programming-books.md (hunk 2 above)
Comment (PedrV) [reply free-programming-books.md:383]: This is an authorized distribution, the second version of the book is now retired and the author made it available in the website cpbook.net, it's easy to check... I just used a different link ...

**Pull request closed without merging**